process/block/shardblock.go

```diff
@@ -922,18 +921,22 @@ func (sp *shardProcessor) updateStateStorage(finalHeaders []data.HeaderHandler)
 			continue
 		}
 
-		rootHash := prevHeader.GetRootHash()
-		if rootHash == nil {
+		prevRootHash := prevHeader.GetRootHash()
```

Comment:
I think you can replace Lines 911-922 with this:

		prevHeader, errNotCritical := process.GetShardHeaderFromStorage(finalHeaders[i].GetPrevHash(), sp.marshalizer, sp.store)
		if errNotCritical != nil {
			log.Debug(errNotCritical.Error())
			continue
		}

Reply:
Done.